### ISSUE TITLE
FreeRDP clipboard issue

### ISSUE THREAD

**Issue by Olmer1st** [FreeRDP/FreeRDP#3309]
Hi,

I have a trouble to make copy-paste on connecting server.
Connecting from Ubuntu 14.04 to CentOS 7.
xfreerdp /v:hostname /u:username /d:domain +clipboard

With mouse select an text in connected machine, make copy... the xfreerdp crashed with errors:
[11:46:32:730] [666:667] [INFO][com.freerdp.client.common.cmdline] - loading channel cliprdr
[11:46:55:978] [666:668] [ERROR][com.freerdp.channels.cliprdr.client] - Invalid short format list length: 24
[11:46:55:978] [666:668] [ERROR][com.freerdp.channels.cliprdr.client] - cliprdr_process_format_list failed with error 1359!
[11:46:55:978] [666:668] [ERROR][com.freerdp.channels.cliprdr.client] - cliprdr_order_recv failed with error 1359!
[11:46:55:978] [666:667] [ERROR][com.freerdp.core] - cliprdr_virtual_channel_client_thread reported an error. Error was 1359
[11:46:55:978] [666:667] [INFO][com.freerdp.client.x11] - Network disconnect!
[11:46:55:978] [666:667] [ERROR][com.freerdp.client.x11] - Failed to check FreeRDP file descriptor

Any idea?
Thx

**Comment by bmiklautz** [comment]
@Olmer1st what version of FreeRDP are you using?

**Comment by Olmer1st** [comment]
This is FreeRDP version 2.0.0-dev (git bd7ed27)
Compiled from git

**Comment by bmiklautz** [comment]
@Olmer1st can you add what host system you are connecting to with xfreerdp?

**Comment by Olmer1st** [comment]
From Ubuntu 14.04   to CentOs 7

**Comment by bmiklautz** [comment]
@Olmer1st let me rephrase this what server are you connecting to? xrdp? VirtualBox?

**Comment by Olmer1st** [comment]
xrdp

**Comment by Olmer1st** [comment]
Hi,
any idea?

**Comment by hardening** [comment]
@Olmer1st would you mind trying with a master nightly build to see if the problem is still present upstream (14.04 is... well... old)

**Comment by Olmer1st** [comment]
@hardening Done. Same behavior.

**Comment by hardening** [comment]
Ok so it seems like xrdp violates the spec in another different way than mstsc and rdpsh (see code to understand)

**Comment by Olmer1st** [comment]
Please note,  it's not happens in version freerdp 1.1.
That mean this bug started from freerdp 2.0....

**Comment by hardening** [comment]
Well you says bug, I'd say we're doing more checks. So a malicious server can not trick us.

**Comment by spimenov** [comment]
wfreerdp.exe (master I got yesterday, built locally with VS2015) was used to connect from Windows 10 to Windows 2012R2:
"wfreerdp.exe /v:<target>" : clipboard doesn't work
"wfreerdp.exe /v:<target> +clipboard" : clipboard works from the target to the host only. Doesn't work other way.
Clipboard was used to copy plain text from notepad to notepad.

**Comment by bmiklautz** [comment]
@spimenov your issue is not related to this one.

**Comment by Olmer1st** [comment]
@hardening  How I can help you? Additional tests?

**Comment by hardening** [comment]
@Olmer1st you could probably also report the bug to the xrdp team, as short formats length is supposed to be a multiple of 36.

**Comment by Olmer1st** [comment]
As I see xrdp teem not so responsive. Might be it can be resolved from your side? Currently I can't work remotely... Or I should disable clipboard synchronization or install older version of FreeRDP, which have other clipboard issues....

**Comment by bmiklautz** [comment]
@Olmer1st I haven't had time to look into this issue in detail yet. However if it is a protocol violation or implementation problem as @hardening stated I don't think it's a good idea to implement a work around for a specific server.

**Comment by Olmer1st** [comment]
@bmiklautz Thanks. Anyway something changed between versions of FreeRDP. xrdp not changed... :(

**Comment by bmiklautz** [comment]
@Olmer1st as described above haven't looked into the details.. but we did a lot of code hardening and added many protocol checks..

**Comment by hardening** [comment]
I will have a deeper look. 
@bmiklautz BTW we're already doing a lot of workaround for ms servers (with violation of their own protocols), so I think we could add one for xrdp...

**Comment by Olmer1st** [comment]
@hardening Thanks!

**Comment by bmiklautz** [comment]
@hardening I don't disagree with you on that but usually they all "violate" the protocol in the same way as in it's not documented properly ;)

**Comment by Olmer1st** [comment]
Hi again,

Anew news about the issue?

Thanks

**Comment by Olmer1st** [comment]
Hi,

Any news?
Thanks

**Comment by bmiklautz** [comment]
@Olmer1st no one has yet looked into it. As this doesn't happen when using FreeRDP against Microsoft servers this isn't high priority at the moment. But be assured it isn't forgotten ;)

**Comment by olberger** [comment]
FWIW, this issue is referred to from https://issues.apache.org/jira/browse/GUACAMOLE-1028, which deals with Guacamole disconnections, that seem to exhibit the current issue.
